p2panda-net/src/sync/manager.rs

```diff
@@ -257,9 +256,15 @@ where
         Ok(())
     }
 
-    /// Remove the given topic from the set of active sync sessions for the given peer. Reschedule
-    /// a sync attempt if the failure was caused by a connection error. Otherwise, drop the attempt
-    /// and schedule the next pending attempt.
+    /// Clean up after a failed sync attempt.
+    ///
+    /// First remove the given topic from the set of active sync sessions for the given peer. Then
+    /// reschedule a sync attempt if the failure was caused by a connection error and the maximum
+    /// number of retry attempts has not already been exceeded. Otherwise, notify the engine actor,
+    /// drop the attempt and schedule the next pending attempt.
```

Comment:
This part kinda repeats what the code describes, I wonder if it's just too much information which will get outdated fast as soon as someone touches the code. I'd try to only write things in comments the code _can't_ convey.